mathics/core/definitions.py

```diff
@@ -494,7 +498,13 @@ def get_definition(self, name, only_if_exists=False) -> "Definition":
         return definition
 
     def get_attributes(self, name):
-        return self.get_definition(name).attributes
+        definition = self.definitions_cache.get(name, None)
```

Comment:
@mmatera I think that caching is not necessary here, it is just acessing a class variable.

Reply:
@TiagoCavalcante, this is an experiment: function calls in Python take 2 to 3 times the time required to access a dictionary by its key. Also, most of the times, (really, 20000 to 1 during doctest) the definition is in the cache, so the call is not necessary.   My hypothesis then is that avoiding calls to get_definition()  when they are not needed could speed up in a constant factor the evaluation. But again, this is just an experiment...

Reply:
@mmatera I'll make some changes tomorrow and put it in another PR. I think this worth dedicated benchmarks.